utilities/transactions/pessimistic_transaction.cc

```diff
@@ -264,7 +264,14 @@ Status PessimisticTransaction::Commit() {
           "Commit-time batch contains values that will not be committed.");
     } else {
       txn_state_.store(AWAITING_COMMIT);
+      if (log_number_ > 0) {
+        dbimpl_->logs_with_prep_tracker()->MarkLogAsHavingPrepSectionFlushed(
+            log_number_);
+      }
       s = CommitWithoutPrepareInternal();
+      if (name_.size() > 0) {
```

Comment:
I don't think that'll be necessarily true. I don't see a reason why transactions can't be named without prepare.

For MyRocks, I think that case may happen, since we'll SetName at the BeginTransaction, since we don't know whether we'll be committing with or without prepare.

Reply:
I stand corrected.